### PR TITLE
chore: upgrade dependencies on 6.x

### PR DESCRIPTION
**Pull request by kacperkapusciak**
Followed https://github.com/react-navigation/react-navigation/pull/11090 and https://github.com/react-navigation/react-navigation/pull/11380

## Test

- `yarn lint`
- `yarn build`
- `yarn test`
- `yarn example test:e2e`

Run example app

### CODE REVIEW

File: packages/native/src/Link.tsx [hunk 1]
```diff
@@ -12,7 +12,10 @@ type Props<ParamList extends ReactNavigation.RootParamList> = {
   onPress?: (
     e: React.MouseEvent<HTMLAnchorElement, MouseEvent> | GestureResponderEvent
   ) => void;
-} & (TextProps & { children: React.ReactNode });
+} & (Omit<TextProps, 'disabled'> & {
+  disabled?: boolean | null;
```
Comment (okwasniewski) [inline packages/native/src/Link.tsx:16]: Why is this change needed? We are adding an option to pass `null`

File: packages/stack/src/types.tsx [hunk 2]
```diff
@@ -396,15 +396,15 @@ export type StackCardInterpolationProps = {
...
   /**
    * Animated node representing whether the card is being swiped (1 - swiping, 0 - not swiping).
    */
-  swiping: Animated.AnimatedInterpolation;
+  swiping: Animated.AnimatedInterpolation<0 | 1>;
   /**
    * Animated node representing multiplier when direction is inverted (-1 - inverted, 1 - normal).
    */
-  inverted: Animated.AnimatedInterpolation;
+  inverted: Animated.AnimatedInterpolation<1 | -1>;
```
Comment (okwasniewski) [inline packages/stack/src/types.tsx:407]:
```suggestion
  inverted: Animated.AnimatedInterpolation<-1 | 1>;
```

Comment (okwasniewski) [reply packages/stack/src/types.tsx:407]: Just to be consistent with other types

**Review by okwasniewski — APPROVED** [review]

**Review by satya164 — APPROVED** [review]

File: packages/native/src/Link.tsx (hunk 1 above)
Comment (kacperkapusciak) [reply packages/native/src/Link.tsx:16]: After the upgrade of `@types/react-native` Typescript started to whine about incompatible `disabled` props:

```
Types of property 'disabled' are incompatible.
    Type 'boolean | null | undefined' is not assignable to type 'boolean | undefined'.
      Type 'null' is not assignable to type 'boolean | undefined'.
``` 

So I've looked up on how it's figured out on `main` and simply did the same:
https://github.com/react-navigation/react-navigation/blob/main/packages/native/src/Link.tsx#L15

File: packages/stack/src/types.tsx (hunk 2 above)
Comment (kacperkapusciak) [reply packages/stack/src/types.tsx]: Thanks! 🙌 
Fixed in [7d7b3f7](https://github.com/react-navigation/react-navigation/pull/11421/commits/7d7b3f7a23dc758866b3c501cb1868b7d09ee8ee)
